### DISCUSSION TITLE
enabling backlog tracing for routes

### DISCUSSION THREAD

**Discussion by divyanshurai07** [apache/camel-karavan#714] [Q&A] [answered]
Hi,

is it possible to add/enable camel backlog tracing for routes so as to help in debugging.

BACKLOGTRACER
Camel supports a backlog tracer interceptor that is used for capturing a trace message of each message as they are routed in Camel. The trace message is stored in a backlog queue, which contains the last N messages for each node in the routes (by default 10).

**Comment by mgubaidullin** [comment, marked as answer]
I believe that you need to add `camel.main.backlog-tracing = true` in `application.properties` to enable backlogtracing

**Comment by davsclaus** [comment]
Camel 3.21 with camel-jbang comes with tracing enabled OOTB and you can use the new `camel trace` command
https://camel.apache.org/manual/camel-jbang.html#_tracing_messages
